tools/cibsecret.c

```diff
@@ -744,8 +733,14 @@ subcommand_check(pcmk__output_t *out, rsh_fn_t rsh_fn, rcp_fn_t rcp_fn,
 
     if ((local_value == NULL) || !pcmk__str_eq(calc_sum, local_sum, pcmk__str_none)) {
         out->err(out, "Checksum mismatch for resource %s parameter %s", rsc, param);
-        *exit_code = CRM_EX_DIGEST;
-        rc = EINVAL;
+
+        /* cibsecret.in would exit with CRM_EX_DIGEST in this case, but no
+         * standard pacemaker return code maps to that (and we don't use
+         * CRM_EX_DIGEST in all that many places anyway).  Rather than add a
+         * new code for just this one place, I'm going to return EIO here and
+         * then manually map that to CRM_EX_DIGEST before we exit.
+         */
+        rc = EIO;
```

Comment:
I don't care enough to hold up the PR unless you want to, but one thing about this does bother me. We don't explicitly return `EIO` anywhere else in `cibsecret.c`, sure. But this assumes we don't get that return code from elsewhere, like `cibsecret`'s calls into `libpacemaker` (and the chained calls into `libcrmcommon`). It's one of those indirect dependencies that could be hard to keep track of down the road, if library return codes ever change.

Reply:
Bah.  There's really not any good option here.  E* is potentially returnable from somewhere else, none of our existing error codes seem to fit, and somehow we don't have any error code that already maps to `CRM_EX_DIGEST`.  It seems like a shame to introduce `pcmk_rc_bad_checksum` just for this, but maybe we would find additional uses for it later.

Reply:
bad_checksum, checksum_mismatch, digest_mismatch, whatever

FWIW, we have another possible use for it in `crm_diff.c:generate_patch()`, but we've chosen to wait until a compatibility break to change the exit code.